Source/JavaScriptCore/runtime/TemporalPlainYearMonth.cpp

```diff
@@ -118,6 +118,105 @@ String TemporalPlainYearMonth::toString(JSGlobalObject* globalObject, JSValue op
     return ISO8601::temporalYearMonthToString(m_plainYearMonth, calendarName);
 }
 
+// https://tc39.es/proposal-temporal/#sec-temporal.plainyearmonth.from
+// https://tc39.es/proposal-temporal/#sec-temporal-totemporalyearmonth
+TemporalPlainYearMonth* TemporalPlainYearMonth::from(JSGlobalObject* globalObject, JSValue itemValue, std::optional<JSValue> optionsValue)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    // Handle string case first so that string parsing errors (RangeError)
+    // can be thrown before options-related errors (TypeError);
+    // see step 4 of ToTemporalYearMonth
+    TemporalPlainYearMonth* result;
+    bool isString = itemValue.isString();
+
+    if (isString) {
+        auto string = itemValue.toWTFString(globalObject);
+        RETURN_IF_EXCEPTION(scope, { });
+        result = TemporalPlainYearMonth::from(globalObject, string);
+        RETURN_IF_EXCEPTION(scope, { });
+        // See step 11 of ToTemporalYearMonth
+        if (optionsValue)
+            toTemporalOverflow(globalObject, optionsValue.value());
+        RETURN_IF_EXCEPTION(scope, { });
+        RELEASE_AND_RETURN(scope, result);
+    }
+
+    std::optional<JSObject*> options;
+    if (optionsValue) {
+        options = intlGetOptionsObject(globalObject, optionsValue.value());
+        RETURN_IF_EXCEPTION(scope, { });
+    }
+
+    if (isString) {
+        // See step 11 of ToTemporalYearMonth
+        // Overflow has to be validated even though it's not used,
+        // so that an error can be thrown for a bad overflow option)
+        if (options) {
+            toTemporalOverflow(globalObject, options.value());
+            RETURN_IF_EXCEPTION(scope, { });
+        }
+        RELEASE_AND_RETURN(scope, { });
+    }
```

Comment:
This may be dead code, I think it's already covered in the earlier `if (isString)` block?